django_components/app_settings.py

```diff
@@ -9,17 +9,24 @@ def __init__(self):
 
     @property
     def AUTODISCOVER(self):
-        return self.settings.setdefault("autodiscover", True)
+        return self.settings.setdefault("AUTODISCOVER", True)
 
     @property
     def LIBRARIES(self):
-        return self.settings.setdefault("libraries", [])
+        return self.settings.setdefault("LIBRARIES", [])
 
     @property
     def TEMPLATE_CACHE_SIZE(self):
-        return self.settings.setdefault("template_cache_size", 128)
+        return self.settings.setdefault("TEMPLATE_CACHE_SIZE", 128)
```

Comment:
This was the result of some detective work. At a certain point I noticed settings weren't being picked up when running tests. It turned out the COMPONENTS dict in django_test_setup.py used uppercase keys. (Just diffed with master to check I'm not making this up.) So that's why I switched to uppercase. Django builtin settings use a mix of uppercase and lowercase in dicts, so in terms of 'idiom' there's probably little difference.

Shall I switch back to lowercase? I'll just lowercase the keys in django_test_setup.py then so settings get picked up.

Reply:
Since there might be existing users with lowercase settings, keeping them lowercase makes the most sense imho. Great catch that django_test_setup used the wrong casing, they should be lowercase too!